node/src/components/consensus/protocols/common.rs

```diff
@@ -17,7 +17,7 @@ pub(crate) fn validators<C: Context>(
     inactive: &HashSet<C::ValidatorId>,
     validator_stakes: BTreeMap<C::ValidatorId, U512>,
 ) -> Validators<C::ValidatorId> {
-    let sum_stakes: U512 = validator_stakes.iter().map(|(_, stake)| *stake).sum();
+    let sum_stakes: U512 = validator_stakes.values().copied().sum();
```

Comment:
Not introduced in this PR but this sum might silently overflow. Is this a concern here?

Reply:
@fizyk20 can you have a look, please?

Reply:
I think we generally aren't worried about overflowing `U512`s. And wouldn't it panic if it overflows, anyway? That's not really a recoverable error, so panicking in that case would be justified in my opinion.

Reply:
It would only panic when it is compiled as debug, in release it will silently overflow. I took a look at the mint code and it does seem that we check for overflow when adjusting the total supply. I guess that the sum of the stakes may never exceed the total supply and then it shouldn't overflow here. Is that a safe assumption?

Reply:
I'm not sure, and when there is doubt, maybe it's better to be on the safe side... :thinking: